python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -1910,10 +1910,17 @@ def can_deserialize(
     def __len__(self) -> int:
         return self.num_partitions
 
-    def __contains__(self, partition_key: str) -> bool:
-        time_window = cast(
-            TimeWindowPartitionsDefinition, self.partitions_def
-        ).time_window_for_partition_key(partition_key)
+    def __contains__(self, partition_key: Optional[str]) -> bool:
+        if partition_key is None:
+            return False
+
+        try:
+            time_window = cast(
+                TimeWindowPartitionsDefinition, self.partitions_def
+            ).time_window_for_partition_key(partition_key)
+        except ValueError:
```

Comment:
is there a reasonable way to check this without exceptions?

Reply:
I assume not but am compelled to ask

Reply:
My worry was that any such attempts there would potentially introduce perf issues (as you'd need to parse the string twice) -- we actually already have some prior art here in the `has_partition_key` method on the PartitionsDefinition class (not the subset), which also uses this sort of sad method.

So perhaps there is some nice way of doing this but it's definitely not immediately obvious